src/bin/uhyve.rs

```diff
@@ -468,15 +477,34 @@ fn run_uhyve() -> i32 {
 
 	let mut app = Args::command();
 	let mut args = Args::parse();
+
 	// Tries to read arguments from a configuration file. If it doesn't exist or if
 	// parsing is not possible, continue using args as-is.
-	//
-	// TODO: Attempt to read configuration file from default locations (cwd, .config, etc.)
-	if let Some(config_file) = args.get_config_file()
-		&& let Ok(contents) = std::fs::read_to_string(config_file)
-		&& let Ok(toml_args) = toml::from_str::<'_, Args>(&contents)
+	if let Some(config_file) = args.get_config_file() {
```

Comment:
Perhaps consider a `for` loop over possible config files.

Reply:
> and happens fairly often, so resolving env vars beforehand should be okay.

yeah it's a micro-optimization at this point but i considered it from the perspective of "what would i do if i were developing minimalist-oriented software" ig.

a bigger problem is that i would have to spend much more time breaking this out into a function because i am not sure if i really want to create an error enum for both reading the file from the filesystem and parsing the toml - this is kinda important for testing the behavior.

Reply:
a loop might not be the best idea 'cause of https://github.com/hermit-os/uhyve/pull/1047#issuecomment-3167154469?

Reply:
you could deal with that by either generating the list of entries to check beforehand (and skipping the config file if necessary) or by performing the check in the loop (given that the loop isn't a hot path anyways, it has at most a dozen iterations)

Reply:
I honestly still do not understand what is the argument for using a loop though. There is different logic for each sequence of steps to determine the presence of a config in each path.

Reply:
I'd be down to clean it up, but I'd rather open an issue and do it later - the addition is now tested as well, paving the way for a refactor.

> but not really necessary.

marking as resolved (and hopefully I'll remember to open a new issue after this is merged)